datatree/datatree.py

```diff
@@ -1345,3 +1345,223 @@ def to_zarr(
 
     def plot(self):
         raise NotImplementedError
+
+    def load(self: DataTree, **kwargs) -> DataTree:
+        """Manually trigger loading of the data referenced by this collection.
+
+
+        End-users generally shouldn't need to call this method directly, since
+        most operations should dispatch to the underlying xarray objects which
+        this collection contains. There may be use cases where a user wants to
+        eagerly load data from disk into memory.
+
+        Parameters
+        ----------
+        **kwargs : dict
+            Additional keyword arguments passed on to ``dask.compute``.
+
+        See Also
+        --------
+        dask.compute
+        """
+        new_datatree_dict = {node.path: node.ds.load(**kwargs) for node in self.subtree}
+        return DataTree.from_dict(new_datatree_dict)
```

Comment:
I'm not sure if this will have the behavior you intend: `DataTree.from_dict` will construct a completely new tree object, and then you are inserting whatever you get when you call `DatasetView.load()`. It's not altering `self` in-place.

Also I think it would be worth double-checking that `DatasetView.load()` does what you expect too with regard to new objects / copying - I never really thought about that case when I wrote `DatasetView`.

If you want to return the same tree but with all the data loaded I think you need to alter the current tree in-place instead of creating a new one, i.e. something like

```python
for node in self.subtree:
    self[node.path] = node.ds.load
```

though this might not fail gracefully...